### DISCUSSION TITLE
How to add a standalone js library into your application like the json-refs.js

### DISCUSSION THREAD

**Discussion by amiths89** [eclipse-theia/theia#14228] [General] [answered]
Am trying to add json-refs dependancy but i came across the fact that would need polyfill and am not very familiar with that. In the jsonforms community they did mention to use the browser build instead of the node install to avoid polyfill. On doing more research i am assuming this needs to be integrated in the package.json. With the reference to the example [https://github.com/eclipse-glsp/glsp-examples/tree/master/project-templates/java-emf-theia](url) am assuming this change needs to go in the package.json in the tasklist-browser-app. Would that be correct and if so can you provide an example of adding a standalone library like that

**Comment by msujew** [comment, marked as answer]
Hey @amiths89,

you should be able to import the browser build of the library using something like:

```
import { ... } from 'json-refs/dist/json-refs';
```

Note that the library is not designed well for use in the browser environment, and is missing the `.d.ts` TypeScript declaration files for that specific import.

**Reply by amiths89** [reply]
oh that sounds simple enough. Thanks @msujew.
